howso/direct/client.py

```diff
@@ -4896,6 +4965,162 @@ def get_trainee_session_training_indices(
             return []
         return result
 
+    def get_hierarchy(self, trainee_id: str) -> Dict:
+        """
+        Output the hierarchy for a trainee.
+
+        Parameters
+        ----------
+        trainee_id : str
+            The ID of the Trainee get hierarchy from.
+
+        Returns
+        -------
+        dict of {str: dict}
+            Dictionary of the currently contained hierarchy as a nested dict
+            with False for trainees that are stored independently.
+        """
+        self._auto_resolve_trainee(trainee_id)
+        return self.howso.get_hierarchy(trainee_id)
+
+    def rename_subtrainee(
+        self,
+        trainee_id: str,
+        new_name: str,
+        *,
+        child_id: Optional[str] = None,
+        child_name_path: Optional[List[str]] = None
+    ) -> None:
+        """
+        Renames a contained child trainee in the hierarchy.
+
+        Parameters
+        ----------
+        trainee_id : str
+            The ID of the Trainee whose child to rename.
+        new_name : str,
+            New name of child trainee
+        child_id : str, optional
+            Unique id of child trainee to rename. Ignored if child_name_path is specified
+        child_name_path : list of str, optional
+            List of strings specifying the user-friendly path of the child
+            subtrainee to rename.
+        """
+        self._auto_resolve_trainee(trainee_id)
+        return self.howso.rename_subtrainee(
+            trainee_id,
+            new_name=new_name,
+            child_id=child_id,
+            child_name_path=child_name_path
+        )
+
+
+    def execute_subtrainee(
```

Comment:
I thought it was decided to remove `execute_subtrainee` from the client? There shouldn't be a method trying to do everything all at once.